packages/core/src/amazonq/webview/ui/tabs/generator.ts

```diff
@@ -39,7 +39,12 @@ export class TabDataGenerator {
         this.highlightCommand = props.commandHighlight
     }
 
-    public getTabData(tabType: TabType, needWelcomeMessages: boolean, taskName?: string): MynahUIDataModel {
+    public getTabData(
+        tabType: TabType,
+        needWelcomeMessages: boolean,
+        taskName?: string,
+        serviceName?: string
```

Comment:
As we discussed, there should be no need to pass this around. Use `isSageMaker('unified')`

Reply:
Why not pass `isSageMakerUnifed: boolean`? Why pass the service name?

Reply:
That kind of logic should not live in the webview. The webview should have minimal business logic.